### ISSUE TITLE
add benchmarks

### ISSUE THREAD

**Issue by pythoncrazy** [Locamage/jimm#16]
compare pytorch vs flax implementations/lines of code

**Comment by pythoncrazy** [comment]
Added profiling to the training loop, it seems to be very good, with little/no bubbles in computation.

**Comment by pythoncrazy** [comment]
am testing on CPU only, since on TPUs the advantage clearly goes to JAX, and do not have GPUs ready for testing

**Comment by pythoncrazy** [comment]
actually this is useless
